servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -407,7 +386,10 @@ private static StreamingHttpClientFilterFactory appendFilter(
         if (currClientFilterFactory == null) {
             return appendClientFilterFactory;
         } else {
-            return client -> currClientFilterFactory.create(appendClientFilterFactory.create(client));
+            return client -> currClientFilterFactory.create(
+                    new ContextAwareDelegateStreamingHttpClient(appendClientFilterFactory.create(client),
+                            ((ContextAwareDelegateStreamingHttpClient) client).lbEventStream(),
+                            ((ContextAwareDelegateStreamingHttpClient) client).sdStatus()));
```

Comment:
With this approach I'm a bit concerned about number of wrapping layers created for each client. Because it will always be on request path, there is a risk it may affect JIT or some other JVM optimizations.

What I had in mind was to extend `StreamingHttpClientFilterFactory` API only for internal purposes:
```java
    @FunctionalInterface
    interface ExtendedStreamingHttpClientFilterFactory extends StreamingHttpClientFilterFactory {

        StreamingHttpClientFilter create(FilterableStreamingHttpClient client, Publisher<Object> lbEventStream, SdStatusCompletable sdStatus);
    }
```
Then we can use it to pass 2 extra arguments:
```java
    private static ExtendedStreamingHttpClientFilterFactory appendExtendedFilter(
            @Nullable final ExtendedStreamingHttpClientFilterFactory currClientFilterFactory,
            final StreamingHttpClientFilterFactory appendClientFilterFactory) {
        if (appendClientFilterFactory instanceof RetryingHttpRequesterFilter) {
            if (currClientFilterFactory == null) {
                return (client, lbEventStream, sdError) -> {
                    ContextAwareRetryingHttpRequesterFilter filter = (ContextAwareRetryingHttpRequesterFilter) ((RetryingHttpRequesterFilter) appendClientFilterFactory).create(client);
                    filter.inject(lbEventStream, sdError);
                    return filter;
                };
            } else {
                return (client, lbEventStream, sdError) -> {
                    ContextAwareRetryingHttpRequesterFilter filter = (ContextAwareRetryingHttpRequesterFilter) ((RetryingHttpRequesterFilter) appendClientFilterFactory).create(client);
                    filter.inject(lbEventStream, sdError);
                    return currClientFilterFactory.create(filter, lbEventStream, sdError);
                };
            }

        } else {
            if (currClientFilterFactory == null) {
                return (client, lbEventStream, sdError) -> appendClientFilterFactory.create(client);
            } else {
                return (client, lbEventStream, sdError) -> currClientFilterFactory.create(appendClientFilterFactory.create(client), lbEventStream, sdError);
            }
        }
    }
```
^ it also creates extra wrappers for `StreamingHttpClientFilterFactory`, but those will be only at the build time.